src/detail/Interpolant.cxx

```diff
@@ -5,16 +5,21 @@
 namespace cubic_splines {
 namespace detail {
 double _find_parameter(std::function<double(double)> const &f,
-                      std::function<double(double)> const &df, double x,
-                      Axis const &axis) {
+                       std::function<double(double)> const &df, Axis const &axis,
+                       double initial_guess) {
   auto low = axis.GetLow();
   auto high = axis.GetHigh();
   auto func = [&f, &df](double x) { return std::make_tuple(f(x), df(x)); };
 
-  auto tol = [](double min, double max) { return (std::abs(max - min) < 1.e-2) ? true : false; };
-  std::tie(low, high) = boost::math::tools::bisect(f, low, high, tol); 
+  if (std::isnan(initial_guess)) {
+    auto tol = [](double min, double max) {
+      return (std::abs(max - min) < 1.e-2) ? true : false;
+    };
+    std::tie(low, high) = boost::math::tools::bisect(f, low, high, tol);
```

Comment:
We need to check whether low==high, otherwise newton_raphson_iterate will throw an error (this happend to me in some cases)

Reply:
I think low should never be the same as high. This must be a corrupt function call and now throw a assert.